### ISSUE TITLE
Improve how to bump the Elastic Stack version

### ISSUE THREAD

**Issue by thbkrkr** [elastic/cloud-on-k8s#3612]
Updating the Elastic Stack version is a manual task prone to omissions. We have a script `hack/update-samples-stack-version.sh` that handles updating the Elastic Stack version only in the samples. Since this script was created, the Elastic Stack version is present in several other places. 

It would be good to improve the existing script to handle updating all Elastic Stack versions present in the repository. We may need to change the current approach and consider adding the old version as a parameter.
The current way is to match `version: [0-9]\.[0-9]\.[0-9]` or `version: "[0-9]\.[0-9]\.[0-9]"` and it does not work for example for `hack/upgrade-test-harness/conf.yaml` which has other old stack versions that should not be updated.

Here are the versions to update:

```
Makefile:368:STACK_VERSION          ?= 7.8.1

test/e2e/test/version.go:19:	LatestVersion7x = "7.8.1" // version to synchronize with the latest release of the Elastic Stack

test/e2e/kb/testdata/kibana_standalone.yaml:7:  version: 7.8.1
test/e2e/kb/testdata/kibana_standalone.yaml:21:  version: 7.8.1

config/e2e/filebeat.yaml:91:          image: docker.elastic.co/beats/filebeat:7.8.1
config/e2e/metricbeat.yaml:134:          image: docker.elastic.co/beats/metricbeat:7.8.1

config/samples/elasticsearch/elasticsearch.yaml:7:  version: 7.8.1
config/samples/apm/apmserver.yaml:6:  version: 7.8.1
config/samples/enterprisesearch/ent_es.yaml:7:  version: 7.8.1
config/samples/enterprisesearch/ent_es.yaml:21:  version: 7.8.1
config/samples/kibana/kibana_es.yaml:7:  version: 7.8.1
config/samples/kibana/kibana_es.yaml:21:  version: 7.8.1
config/samples/apm/apm_es_kibana.yaml:8:  version: 7.8.1
config/samples/apm/apm_es_kibana.yaml:22:  version: 7.8.1
config/samples/apm/apm_es_kibana.yaml:32:  version: 7.8.1

config/recipes/associations-rbac/apm_es_kibana_rbac.yaml:87:  version: 7.8.1
config/recipes/associations-rbac/apm_es_kibana_rbac.yaml:100:  version: 7.8.1
config/recipes/associations-rbac/apm_es_kibana_rbac.yaml:114:  version: 7.8.1
config/recipes/istio-gateway/03-elasticsearch-kibana.yaml:10:  version: 7.8.1
config/recipes/istio-gateway/03-elasticsearch-kibana.yaml:100:  version: 7.8.1
config/recipes/gclb/01-elastic-stack.yaml:9:  version: 7.8.1
config/recipes/gclb/01-elastic-stack.yaml:48:  version: 7.8.1
config/recipes/beats/heartbeat_es_kb_health.yaml:7:  version: 7.8.1
config/recipes/beats/heartbeat_es_kb_health.yaml:30:  version: 7.8.1
config/recipes/beats/heartbeat_es_kb_health.yaml:42:  version: 7.8.1
config/recipes/beats/stack_monitoring.yaml:9:  version: 7.8.1
config/recipes/beats/stack_monitoring.yaml:121:  version: 7.8.1
config/recipes/beats/stack_monitoring.yaml:213:  version: 7.8.1
config/recipes/beats/stack_monitoring.yaml:232:  version: 7.8.1
config/recipes/beats/stack_monitoring.yaml:250:  version: 7.8.1
config/recipes/beats/stack_monitoring.yaml:262:  version: 7.8.1
config/recipes/beats/filebeat_autodiscover.yaml:7:  version: 7.8.1
config/recipes/beats/filebeat_autodiscover.yaml:97:  version: 7.8.1
config/recipes/beats/filebeat_autodiscover.yaml:109:  version: 7.8.1
config/recipes/beats/filebeat_autodiscover_by_metadata.yaml:7:  version: 7.8.1
config/recipes/beats/filebeat_autodiscover_by_metadata.yaml:99:  version: 7.8.1
config/recipes/beats/filebeat_autodiscover_by_metadata.yaml:111:  version: 7.8.1
config/recipes/beats/auditbeat_hosts.yaml:7:  version: 7.8.1
config/recipes/beats/auditbeat_hosts.yaml:79:        #  image: docker.elastic.co/beats/auditbeat:7.8.1
config/recipes/beats/auditbeat_hosts.yaml:121:  version: 7.8.1
config/recipes/beats/auditbeat_hosts.yaml:133:  version: 7.8.1
config/recipes/beats/metricbeat_hosts.yaml:7:  version: 7.8.1
config/recipes/beats/metricbeat_hosts.yaml:172:  version: 7.8.1
config/recipes/beats/metricbeat_hosts.yaml:184:  version: 7.8.1
config/recipes/beats/packetbeat_dns_http.yaml:7:  version: 7.8.1
config/recipes/beats/packetbeat_dns_http.yaml:47:  version: 7.8.1
config/recipes/beats/packetbeat_dns_http.yaml:59:  version: 7.8.1
config/recipes/beats/journalbeat_hosts.yaml:7:  version: 7.8.1
config/recipes/beats/journalbeat_hosts.yaml:52:  version: 7.8.1
config/recipes/beats/journalbeat_hosts.yaml:64:  version: 7.8.1
config/recipes/beats/filebeat_no_autodiscover.yaml:7:  version: 7.8.1
config/recipes/beats/filebeat_no_autodiscover.yaml:56:  version: 7.8.1
config/recipes/beats/filebeat_no_autodiscover.yaml:68:  version: 7.8.1
config/recipes/traefik/02-elastic-stack.yaml:9:  version: 7.8.1
config/recipes/traefik/02-elastic-stack.yaml:59:  version: 7.8.1
config/recipes/traefik/02-elastic-stack.yaml:71:  version: 7.8.1
config/recipes/apm-server-jaeger/apm-server-jaeger.yaml:7:  version: 7.8.1

hack/upgrade-test-harness/conf.yaml:16:    stackVersion: 7.8.1
hack/upgrade-test-harness/conf.yaml:20:    stackVersion: 7.8.1
hack/upgrade-test-harness/testdata/upcoming/stack.yaml:7:  version: 7.8.1
hack/upgrade-test-harness/testdata/upcoming/stack.yaml:19:  version: 7.8.1
hack/upgrade-test-harness/testdata/upcoming/stack.yaml:29:  version: 7.8.1
hack/upgrade-test-harness/testdata/v112/stack.yaml:7:  version: 7.8.1
hack/upgrade-test-harness/testdata/v112/stack.yaml:19:  version: 7.8.1
hack/upgrade-test-harness/testdata/v112/stack.yaml:29:  version: 7.8.1
```

**Comment by anyasabo** [comment]
I think we also may want to consider `.ci/packer_cache.sh`

**Comment by thbkrkr** [comment]
Closed in #4385.
